### PR TITLE
Use BCEWithLogitsLoss as many as possible for AMP

### PR DESCRIPTION
**Pull request by gaotongxiao**
And update type hints in BaseTextDetPostprocessor

### CODE REVIEW

File: mmocr/models/textdet/heads/db_head.py [hunk 1]
```diff
@@ -103,3 +84,101 @@ def _init_thr(self,
             nn.BatchNorm2d(inner_channels // 4), nn.ReLU(inplace=True),
             nn.ConvTranspose2d(inner_channels // 4, 1, 2, 2), nn.Sigmoid())
         return seq
+
+    def forward(self,
+                img: Tensor,
+                data_samples: Optional[List[TextDetDataSample]] = None,
+                mode: str = 'loss') -> Tuple[Tensor, Tensor, Tensor]:
```
Comment (Harold-lkk) [inline mmocr/models/textdet/heads/db_head.py:91]:
```suggestion
                mode: str = 'predict') -> Tuple[Tensor, Tensor, Tensor]:
```
better for inferencing

File: mmocr/models/textdet/heads/db_head.py [hunk 2]
```diff
@@ -103,3 +84,101 @@ def _init_thr(self,
...
+        prob_logits = self.binarize(img).squeeze(1)
+        prob_map = prob_logits.sigmoid()
+        if mode == 'predict':
+            return prob_map
+        thr_map = self.threshold(img).squeeze(1)
+        binary_map = self._diff_binarize(
+            prob_logits.sigmoid(), thr_map, k=50).squeeze(1)
+        if mode == 'loss':
+            return prob_logits, thr_map, binary_map
+        return prob_logits, thr_map, binary_map, prob_map
```
Comment (Harold-lkk) [inline mmocr/models/textdet/heads/db_head.py:121]: self.pred_sigmoid = nn.Sigmoid()
```suggestion
        prob_map = self.pred_sigmoid(prob_logits)
        if mode == 'predict':
            return prob_map
        thr_map = self.threshold(img).squeeze(1)
        binary_map = self._diff_binarize(
            prob_map, thr_map, k=50).squeeze(1)
        if mode == 'loss':
            return prob_logits, thr_map, binary_map
        return prob_logits, thr_map, binary_map, prob_map
```

**Review by Harold-lkk — APPROVED** [review]
